### ISSUE TITLE
Can't create Amazon EC2 IO1 volume

### ISSUE THREAD

**Issue by christianbundy** [docker/machine#3189]
When you set `--amazonec2-volume-type io1`, it errors out:

>Error creating machine: Error in driver during machine creation: Error launching instance: InvalidParameterCombination: The parameter iops must be specified for io1 volumes.

**Comment by etano** [comment]
+1

**Comment by fgassert** [comment]
Would be nice to have provisioned IOPS flag. Not many other options for disk-constrained processes, other than spinning up machine yourself and manually setting it up.

**Comment by chillosu** [comment]
Any plans to implement this? We use a high amount of iops1 but don't require a lot of space in our auto-scaled runners.

**Comment by MattHoy** [comment]
I was able to patch this myself (Lazily) by adding this line:
                      Iops:                aws.Int64(2000),
to this file: https://github.com/docker/machine/blob/master/drivers/amazonec2/amazonec2.go

on line: 616

Then recompiling as per the instructions here: https://github.com/docker/machine/blob/master/CONTRIBUTING.md

This works, but it means it will always be set and you MUST use io1 as the volume type (Or any other volume type that takes an IOPS parameter). This met our use case so we were fine with it.

**Comment by mgamsjager** [comment]
Is this actual a planned feature?

**Comment by kamadorueda** [comment]
would be really awesome, downloading big docker layers in a gp2 volume type is pretty slow, not because of bandwidth, but because the 'extracting' stage

**Comment by codinghedgehog-python** [comment]
Searching out this error brought me here. Is there any resolution to this ? Can the `iops` parameter to be specified in any way from cli ?

**Comment by mrsemmix** [comment]
was this every solved ?

**Comment by codinghedgehog-python** [comment]
@Semmix  I don't think so 'coz I even asked people on StackOverflow and nobody chimed in despite the question had dozen views. Personally, I attached the volume manually from the web interface and I'm controlling the instance through SSH. It's far from ideal but as of now Docker-Machine still has this limitation. It makes me wonder though what other developers do when they work with Docker Machine and need to attach different volume. Do they all do what I did or they came up with better workaround .....

**Comment by mrsemmix** [comment]
i used this : Option to allow user defined IOPS #4771
Had to recompile it on the target machine, it unbelievable that such an easy fix is being overlooked for over 2 years. i assume its just not maintained anymore or something.

**Comment by codinghedgehog-python** [comment]
> i used this : Option to allow user defined IOPS #4771
> Had to recompile it on the target machine, it unbelievable that such an easy fix is being overlooked for over 2 years. i assume its just not maintained anymore or something.
@Semmix 
So what exactly did you do ? Did you build the docker-machine from source instead of pulling it from the official, public repo ?

I tried to pull latest stable release which is v0.16.2 from the public repo and when I ran:
`docker-machine create -d amazonec2 --amazonec2-secret-key abc --amazonec2-access-key def --amazonec2-region eu-west-2 --amazonec2-volume-type "io1" --device-read-iops 300 --amazonec2-instance-type "t2.small" instance_name`

I got:
`flag provided but not defined: -device-read-iops`

So how did you make it work for you ?

**Comment by codinghedgehog-python** [comment]
> i used this : Option to allow user defined IOPS #4771
> Had to recompile it on the target machine, it unbelievable that such an easy fix is being overlooked for over 2 years. i assume its just not maintained anymore or something.

@Semmix  

I'm doing it now. When I run `make` it exits out with:

`rm -Rf /home/utcloud/docker-machine/bin/*`

Same output with `sudo`

... meaning it's trying to delete contents of the `bin` directory but it's failing for some reason. How did you get around it ?
